lib/compilers/sway.ts

```diff
@@ -233,25 +237,34 @@ export class SwayCompiler extends BaseCompiler {
     }
 }
 
-const FORC_TOML_CONTENT = `[project]
-entry = "main.sw"
-license = "Apache-2.0"
-name = "compiler-explorer"
-
-[dependencies]
-`;
-
 async function setupForcProject(
     projectDir: string,
     inputFilename: string,
+    compilerVersion: string,
 ): Promise<{mainSw: string; symbolsPath: string}> {
     const outDebugDir = path.join(projectDir, 'out', 'debug');
     const symbolsPath = path.join(outDebugDir, 'symbols.json');
     await fs.mkdirp(outDebugDir);
 
+    // Create base Forc.toml content
+    let forcTomlContent = `[project]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "compiler-explorer"
+
+[dependencies]`;
+
+    // In local development, the compiler version is set to 'latest' and should use local dependencies.
+    // On Compiler Explorer servers, we have specific versioned compilers (e.g., "0.66.7") with
+    // matching pre-installed std libraries that we need to explicitly reference in the Forc.toml.
+    if (compilerVersion !== 'latest') {
+        forcTomlContent += `
+    std = { path = "/opt/compiler-explorer/libs/sway/std/v${compilerVersion}" }`;
```

Comment:
@mattgodbolt Are you able to advise what the correct path should be here to access the lib relative to the local project dir?

Reply:
That looks good: and I updated the infra PR accordingly:

```
Feb/26 17:30 admin-node~/infra (main|✔) 3.6m $ ls -l /opt/compiler-explorer/libs/sway/std/
total 4
drwxr-xr-x 30 root root 6144 Feb 26 17:29 v0.66.7
```

Hard-coding the path (`/opt/compiler-explorer/libs/sway/std/...` seems sad, but I don't know if anyone will ever run this locally with a non-default installation.

Reply:
I'm a bit confused, you've added it as a `lib` as well? Will the user pick this?

I'd suggest adding a "std" type field to the compiler and looking this up. then adding:

```
compiler.sway0667.std=/opt/compiler-explorer/libs/sway/std/v0.66.7
```

or whatever, and look that up here, if set. Then you won't even need to `compilerVersion != 'latest'` as the "should I add std" comes from whether the option is set or not.
\
I hope that makes sense?

Reply:
Agh nice this is much cleaner! Changed in [e2d9eac](https://github.com/compiler-explorer/compiler-explorer/pull/7456/commits/e2d9eacbbe3f5e03f5908655673f3840191a6197)